### PR TITLE
Bh/1403 repeat writer view

### PR DESCRIPTION
**Pull request by billyhunt**
[Link to ZenHub issue.](https://github.com/usdoj-crt/crt-portal-management/issues/1403)

## What does this change?

Improve performance by creating a table to check for repeat writer info.  This table is keyed off of email vs report, so existing emails can be checked for new reports, without requiring any real time updates.

## Checklist:

+ [x] Check form/view and see if #total count is "-" for a reports with no email and 1 for ones with an email.

+ [x] run  ```docker-compose run web python /code/crt_portal/manage.py generate_repeat_writer_info```
+ [x] Make sure you have some repeat writers in your DB, and check those to see if you have an accurate count for all reports.
+ [x] Change a report to add an email that has been repeated elsewhere. Check to see that after the change, the count is NOT updated.
+ [x] rerun ```docker-compose run web python /code/crt_portal/manage.py generate_repeat_writer_info```, the count should update.
+ [x] go to a report detail on a report with a repeat writer, make sure the count is ok. 

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [x] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/views.py [hunk 1]
```diff
@@ -226,6 +227,17 @@ def index_view(request):
...
             report.other_class,
         )
+        # If a user has an email, it is looked up in the table to see if they are a repeat writer.
+        if report.contact_email:
+            repeat_writer = RepeatWriterInfo.objects.filter(email=report.contact_email).first()
+            if repeat_writer:
+                report.related_reports_count = repeat_writer.email_count
+            # If the email is not in the repeat_writer table, add it and initialize to 1.
+            else:
+                report.related_reports_count = 1
```
Comment (JackRyan1989) [inline crt_portal/cts_forms/views.py:237]: @billyhunt Why is the default value 1? There are no related reports if they aren't a repeat writer, so shouldn't it be 0?

**Review by JackRyan1989 — COMMENTED** [review]
Looks good, just one comment/question.

File: crt_portal/cts_forms/views.py (hunk 1 above)
Comment (JackRyan1989) [reply crt_portal/cts_forms/views.py:237]: Okey dokey. Yeah that makes the most sense. Keep it consistent and effectively related reports = total reports. Thanks!

**Review by JackRyan1989 — APPROVED** [review]
Looks good to me!

**Comment by billyhunt** [comment]
@ilodidoj Updates to make this case insensitive have been added, and this is ready to be reviewed.

**Review by ilodidoj — COMMENTED** [review]
@billyhunt @JackRyan1989 this may have some more room for optimization by going to materialized view route.  We should not default to 1, '-' used for records where there is no email even if it is by same person.  This is accurate representation of email count validated each existing report.  Let's discuss this more during a pairing session.

**Comment by billyhunt** [comment]
@JackRyan1989 and @ilodidoj This now is using a materialized view instead of a table.  Could you all re-review?

**Review by ilodidoj — CHANGES_REQUESTED** [review]
@billyhunt please check the circleci config.yml.  "python crt_portal/manage.py generate repeat_writer_info" --name generate repeat_writer_info << '_' underscore missing for generate and name should be formatted with dash - not underscore _ to be consistant. Should not have any space between generate and repeat, need ' _ ' for the command and '-' for the name following 'generate'.

**Comment by JackRyan1989** [comment]
New changes check out. Looks good to me!

**Comment by ilodidoj** [comment]
@billyhunt correspondence information on detail page not showing the view all link even if the email count is more than 1.  Ref: https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/cts_forms/templates/forms/complaint_view/show/correspondent_info.html line 35.  Please check.

**Review by ilodidoj — APPROVED** [review]

**Comment by ilodidoj** [comment]
@billyhunt count now shows in both places. Working as expected. Approved and ready to merge in dev.

**Pull request merged**